pkg/server/server.go

```diff
@@ -102,6 +102,13 @@ type ProxyServer struct {
 	// BackendManager manages the backends.
 	BackendManager BackendManager
 
+	// Readiness reports if the proxy server is ready, i.e., if the proxy
+	// server has connections to proxy agents (backends). Note that the
+	// proxy server does not check the healthiness of the connections,
+	// though the proxy agents do, so this readiness check might report
+	// ready but there is no healthy connection.
+	Readiness ReadinessManager
```

Comment:
Not needed yet but with most K/K liveness/readiness checks we provide a mechanism to disable names checks. I'm guessing we will eventually want this functionality.